lib/cext/ruby/ruby.h

```diff
@@ -541,7 +546,7 @@ volatile VALUE *rb_gc_guarded_ptr_val(volatile VALUE *ptr, VALUE val);
 #endif
 
 #ifndef RB_GC_GUARD
-#define RB_GC_GUARD(v) (v)
+#define RB_GC_GUARD(v) (*RB_GC_GUARD_PTR(&(v)))
```

Comment:
Mmh, defining RB_GC_GUARD like this does not work as it would convert to native.
Maybe we can do a `truffle_invoke` passing the value to Java, and there passing it behind a TruffleBoundary so that we ensure the value cannot be GC'd until after the `RB_GC_GUARD`?
Not exactly optimal but sounds safer.
cc @aardvark179